src/Aggregations/DateHistogramAggregation.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Spatie\ElasticsearchQueryBuilder\Aggregations;
+
+class DateHistogramAggregation extends Aggregation
+{
+    protected string $field;
+    protected string $calendarInterval;
+
+    public static function create(string $name, string $field, string $calendarInterval): self
+    {
+        return new self($name, $field, $calendarInterval);
+    }
+
+    public function __construct(string $name, string $field, string $calendarInterval)
+    {
+        $this->name = $name;
+        $this->field = $field;
+        $this->calendarInterval = $calendarInterval;
+    }
+
+    public function payload(): array
+    {
+        $parameters = [
+            'field' => $this->field,
+            'calendar_interval' => $this->calendarInterval,
+        ];
+
+        return [
+            'date_histogram' => $parameters,
+        ];
+    }
```

Comment:
I've used your suggestion in the application I'm working on, but I needed an aggregation in the histogram. See the suggestion for how I've implemented the aggregations.

```suggestion
        public function __construct(
        string $name,
        string $field,
        string $calendarInterval,
        Aggregation ...$aggregations
    ) {
        $this->name = $name;
        $this->field = $field;
        $this->calendarInterval = $calendarInterval;
        $this->aggregations = new AggregationCollection(...$aggregations);
    }

    public static function create(string $name, string $field, string $calendarInterval, Aggregation ...$aggregations): self
    {
        return new self($name, $field, $calendarInterval, ...$aggregations);
    }

    public function payload(): array
    {
        $payload = [
            'date_histogram' => [
                'field' => $this->field,
                'calendar_interval' => $this->calendarInterval,
            ],
        ];

        if (! $this->aggregations->isEmpty()) {
            $payload['aggs'] = $this->aggregations->toArray();
        }

        return $payload;
    }
```